File: easy/multiples.pl
use strict;
use POSIX;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    next if ( $line =~ m/^\s$/ );
    my ( $n, $m ) = ( $line =~ /(\d+),(\d+)/ );
    my $c =
      $n - floor( $n * 2**( -log($m) / log(2) ) ) * 2**( log($m) / log(2) );
    printf( "%d\n", ( $c > 0 ) ? $n - $c + $m : $n );
}
close(INFILE);

File: easy/multiples.rb
File.open(ARGV[0]).each_line do |line|
  s = line.split(',').map(&:to_i)
  c = s[0] - (s[0] * 2**-Math.log2(s[1])).floor * 2**Math.log2(s[1])
  puts c > 0 ? s[0] - c.to_i + s[1] : s[0]
end
